presto-kinesis/src/main/java/io/prestosql/plugin/kinesis/KinesisConnectorFactory.java

```diff
@@ -0,0 +1,168 @@
+import io.prestosql.spi.type.TypeManager;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Supplier;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * This factory class creates the KinesisConnector during server start and binds all the dependency
+ * by calling create() method.
+ */
+public class KinesisConnectorFactory
+        implements ConnectorFactory
+{
+    public static final String connectorName = "kinesis";
+    private static final Logger log = Logger.get(KinesisConnectorFactory.class);
+
+    private final ClassLoader classLoader;
+    private TypeManager typeManager;
+    private NodeManager nodeManager;
+    private Optional<Supplier<Map<SchemaTableName, KinesisStreamDescription>>> tableDescriptionSupplier = Optional.empty();
+    private Map<String, String> optionalConfig = ImmutableMap.of();
+    private Optional<Class<? extends KinesisClientProvider>> altProviderClass = Optional.empty();
+
+    private KinesisHandleResolver handleResolver;
+
+    private Injector injector;
+
+    KinesisConnectorFactory(ClassLoader classLoader, Optional<Supplier<Map<SchemaTableName, KinesisStreamDescription>>> tableDescriptionSupplier,
+            Map<String, String> optionalConfig,
+            Optional<Class<? extends KinesisClientProvider>> altProviderClass)
+    {
+        this.classLoader = classLoader;
+        this.tableDescriptionSupplier = requireNonNull(tableDescriptionSupplier, "tableDescriptionSupplier is null");
+        this.optionalConfig = requireNonNull(optionalConfig, "optionalConfig is null");
+        this.altProviderClass = requireNonNull(altProviderClass, "altProviderClass is null");
+
+        this.handleResolver = new KinesisHandleResolver(connectorName);
+
+        // Explanation: AWS uses a newer version of jackson (2.6.6) than airlift (2.4.4).  In order to upgrade
+        // to the latest version of the AWS API, we need to turn this feature off.  This can be set
+        // in jvm.properties but trying to make this more foolproof.
+        System.setProperty("com.amazonaws.sdk.disableCbor", "true");
+    }
+
+    @Override
+    public String getName()
+    {
+        return connectorName;
+    }
+
+    @Override
+    public ConnectorHandleResolver getHandleResolver()
+    {
+        return new KinesisHandleResolver(connectorName);
+    }
+
+    @Override
+    public Connector create(String connectorId, Map<String, String> config, ConnectorContext context)
+    {
+        log.info("In connector factory create method.  Connector id: " + connectorId);
+        requireNonNull(connectorId, "connectorId is null");
+        requireNonNull(config, "config is null");
+
+        try {
+            Bootstrap app = new Bootstrap(
+                    new JsonModule(),
+                    new KinesisConnectorModule(),
+                    binder -> {
+                        binder.bindConstant().annotatedWith(Names.named("connectorId")).to(connectorId);
```

Comment:
Connector ID is no longer required in connectors. You should be able to remove this everywhere.

Reply:
removed